### ISSUE TITLE
Upgrading to 1.5.0 causes unexpected error when calling predict endpoint of Python custom model

### ISSUE THREAD

**Issue by FarrandTom** [SeldonIO/seldon-core#2786]
## Describe the bug

I am deploying a very simple custom wrapped model written in Python. All the model class does is take an arbitrary JSON list and returns `[0.5]`. 

This works as expected with Core 1.4.0, however the error below is introduced when upgrading to Core 1.5.0. This despite following the outlined upgrading procedure (namely, modifying the Dockerfile to remove the `$API_TYPE` of `REST` specification). 

The error returned when running with Core 1.5.0 is as follows:
```
(seldon-demos) tom@thinkpad:~/Github/personalization-seldon-poc-master$ python container_test.py 
2020-12-16 14:04:55.456580: W tensorflow/stream_executor/platform/default/dso_loader.cc:59] Could not load dynamic library 'libcudart.so.10.1'; dlerror: libcudart.so.10.1: cannot open shared object file: No such file or directory
2020-12-16 14:04:55.456606: I tensorflow/stream_executor/cuda/cudart_stub.cc:29] Ignore above cudart dlerror if you do not have a GPU set up on your machine.
Traceback (most recent call last):
  File "/home/tom/miniconda3/envs/seldon-demos/lib/python3.6/site-packages/urllib3/connectionpool.py", line 677, in urlopen
    chunked=chunked,
  File "/home/tom/miniconda3/envs/seldon-demos/lib/python3.6/site-packages/urllib3/connectionpool.py", line 426, in _make_request
    six.raise_from(e, None)
  File "<string>", line 3, in raise_from
  File "/home/tom/miniconda3/envs/seldon-demos/lib/python3.6/site-packages/urllib3/connectionpool.py", line 421, in _make_request
    httplib_response = conn.getresponse()
  File "/home/tom/miniconda3/envs/seldon-demos/lib/python3.6/http/client.py", line 1379, in getresponse
    response.begin()
  File "/home/tom/miniconda3/envs/seldon-demos/lib/python3.6/http/client.py", line 311, in begin
    version, status, reason = self._read_status()
  File "/home/tom/miniconda3/envs/seldon-demos/lib/python3.6/http/client.py", line 293, in _read_status
    raise BadStatusLine(line)
http.client.BadStatusLine: @@ ?

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/tom/miniconda3/envs/seldon-demos/lib/python3.6/site-packages/requests/adapters.py", line 449, in send
    timeout=timeout
  File "/home/tom/miniconda3/envs/seldon-demos/lib/python3.6/site-packages/urllib3/connectionpool.py", line 727, in urlopen
    method, url, error=e, _pool=self, _stacktrace=sys.exc_info()[2]
  File "/home/tom/miniconda3/envs/seldon-demos/lib/python3.6/site-packages/urllib3/util/retry.py", line 403, in increment
    raise six.reraise(type(error), error, _stacktrace)
  File "/home/tom/miniconda3/envs/seldon-demos/lib/python3.6/site-packages/urllib3/packages/six.py", line 734, in reraise
    raise value.with_traceback(tb)
  File "/home/tom/miniconda3/envs/seldon-demos/lib/python3.6/site-packages/urllib3/connectionpool.py", line 677, in urlopen
    chunked=chunked,
  File "/home/tom/miniconda3/envs/seldon-demos/lib/python3.6/site-packages/urllib3/connectionpool.py", line 426, in _make_request
    six.raise_from(e, None)
  File "<string>", line 3, in raise_from
  File "/home/tom/miniconda3/envs/seldon-demos/lib/python3.6/site-packages/urllib3/connectionpool.py", line 421, in _make_request
    httplib_response = conn.getresponse()
  File "/home/tom/miniconda3/envs/seldon-demos/lib/python3.6/http/client.py", line 1379, in getresponse
    response.begin()
  File "/home/tom/miniconda3/envs/seldon-demos/lib/python3.6/http/client.py", line 311, in begin
    version, status, reason = self._read_status()
  File "/home/tom/miniconda3/envs/seldon-demos/lib/python3.6/http/client.py", line 293, in _read_status
    raise BadStatusLine(line)
urllib3.exceptions.ProtocolError: ('Connection aborted.', BadStatusLine('\x00\x00\x18\x04\x00\x00\x00\x00\x00\x00\x04\x00@\x00\x00\x00\x05\x00@\x00\x00\x00\x06\x00\x00 \x00þ\x03\x00\x00\x00\x01\x00\x00\x04\x08\x00\x00\x00\x00\x00\x00?\x00\x01',))

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "container_test.py", line 9, in <module>
    method='predict')
  File "/home/tom/miniconda3/envs/seldon-demos/lib/python3.6/site-packages/seldon_core/seldon_client.py", line 682, in microservice
    return microservice_api_rest_seldon_message(**k)
  File "/home/tom/miniconda3/envs/seldon-demos/lib/python3.6/site-packages/seldon_core/seldon_client.py", line 842, in microservice_api_rest_seldon_message
    data={"json": json.dumps(payload)},
  File "/home/tom/miniconda3/envs/seldon-demos/lib/python3.6/site-packages/requests/api.py", line 119, in post
    return request('post', url, data=data, json=json, **kwargs)
  File "/home/tom/miniconda3/envs/seldon-demos/lib/python3.6/site-packages/requests/api.py", line 61, in request
    return session.request(method=method, url=url, **kwargs)
  File "/home/tom/miniconda3/envs/seldon-demos/lib/python3.6/site-packages/requests/sessions.py", line 530, in request
    resp = self.send(prep, **send_kwargs)
  File "/home/tom/miniconda3/envs/seldon-demos/lib/python3.6/site-packages/requests/sessions.py", line 643, in send
    r = adapter.send(request, **kwargs)
  File "/home/tom/miniconda3/envs/seldon-demos/lib/python3.6/site-packages/requests/adapters.py", line 498, in send
    raise ConnectionError(err, request=request)
requests.exceptions.ConnectionError: ('Connection aborted.', BadStatusLine('\x00\x00\x18\x04\x00\x00\x00\x00\x00\x00\x04\x00@\x00\x00\x00\x05\x00@\x00\x00\x00\x06\x00\x00 \x00þ\x03\x00\x00\x00\x01\x00\x00\x04\x08\x00\x00\x00\x00\x00\x00?\x00\x01',))
```

## To reproduce

1. Download and unzip the attached model code. 
2. Change dir into the model code. 
3. Run `docker build . -t adusa/azure_function_testing`
4. Run `docker run --name "azure_function_testing" -p 5001:5000 adusa/azure_function_testing`
5. Call your newly deployed container image, with `python container_test.py`
6. You will see the error as above.

## Expected behaviour

The model should return the following 200 response: 
```
meta {
}
data {
  tensor {
    shape: 1
    values: 0.5
  }
}
```

## Environment

You can re-create the environment using the attached .zip of the code. 

## Model Details 
N/A the model is not a predictor. 

Model code: 
[personalization-seldon-poc-master.zip](https://github.com/SeldonIO/seldon-core/files/5703233/personalization-seldon-poc-master.zip)

**Comment by cliveseldon** [comment]
Is this because you are hitting the grpc endpoint as both are available in 1.5.0?

**Comment by FarrandTom** [comment]
Ah ok, that makes sense. The REST endpoint defaults to running on port 9000, whereas the gRPC one is port 5000 it seems. Thanks @cliveseldon!
